### ISSUE TITLE
Compilers should have a public API

### ISSUE THREAD

**Issue by sjrd** [sbt/zinc#124]
https://github.com/sbt/zinc/blob/2529bbd58e458bc7c09fa1931faa09559e584372/internal/compiler-interface/src/main/java/xsbti/compile/Compilers.java

The interface is empty. It seems the only existing way to use it is like at https://github.com/sbt/zinc/blob/9b61c802f7c110deb4263a5971c07196746051a0/zinc/src/main/scala/sbt/internal/inc/IncrementalCompilerImpl.scala#L32-L41, which is clearly indicated as hackery.

The methods `javac()` and `scalac()` should be made publicly available.

**Comment by sjrd** [comment]
Well ... It's worse than I thought, because the `case class Compilers` is `private[sbt]`, defined at https://github.com/sbt/zinc/blob/9b61c802f7c110deb4263a5971c07196746051a0/zinc/src/main/scala/sbt/internal/inc/IncrementalCompilerImpl.scala#L206

So ... This is completely inaccessible. To get to `compilers.scalac.scalaInstance`, I currently have to use reflection:

``` scala
type CompilersAccess = {
  def scalac(): xsbti.compile.CachedCompilerProvider
}

val compilers = (compileInputs in compile).value.compilers
val scalaInstance = compilers.asInstanceOf[CompilersAccess].scalac().scalaInstance
```

**Comment by eed3si9n** [comment]
Fixed in #128 and #132
